internal/handler/handlers.go

```diff
@@ -96,13 +96,25 @@ func DevelopersHandler(svr server.Server, devRepo *developer.Repository) http.Ha
 
 func SubmitDeveloperProfileHandler(svr server.Server, devRepo *developer.Repository) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		svr.RenderPageForProfileRegistration(w, r, devRepo, "submit-developer-profile.html")
+		profile, _ := middleware.GetUserFromJWT(r, svr.SessionStore, svr.GetJWTSigningKey())
+		if profile != nil {
+			routeName := fmt.Sprintf("%s-Developers", strings.Title(svr.GetConfig().SiteJobCategory))
+			svr.Redirect(w, r, http.StatusMovedPermanently, fmt.Sprintf("https://%s/%s", svr.GetConfig().SiteHost, routeName))
+		} else {
+			svr.RenderPageForProfileRegistration(w, r, devRepo, "submit-developer-profile.html")
+		}
 	}
 }
 
 func SubmitRecruiterProfileHandler(svr server.Server, devRepo *developer.Repository) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		svr.RenderPageForProfileRegistration(w, r, devRepo, "submit-recruiter-profile.html")
+		profile, _ := middleware.GetUserFromJWT(r, svr.SessionStore, svr.GetJWTSigningKey())
+		if profile != nil {
+			routeName := fmt.Sprintf("%s-Developers", strings.Title(svr.GetConfig().SiteJobCategory))
+			svr.Redirect(w, r, http.StatusMovedPermanently, fmt.Sprintf("https://%s/%s", svr.GetConfig().SiteHost, routeName))
```

Comment:
can you please remove the else and return early here? to avoid unnecessary nesting
also please use URLProtocol if you can and don't hardcode https

Reply:
same for line 102